src/rendering/htmlformats.jl

```diff
@@ -0,0 +1,143 @@
+# HTML
+# ----
+
+abstract type HTMLFormat <: WeaveFormat end
+
+Base.@kwdef mutable struct JMarkdown2HTML <: HTMLFormat
+    description = "Julia markdown to html"
+    codestart = "\n"
+    codeend = "\n"
+    outputstart = "<pre class=\"output\">"
+    outputend = "</pre>\n"
+    fig_ext = ".png"
+    mimetypes = ["image/png", "image/jpg", "image/svg+xml",
+                "text/html", "text/markdown", "text/plain"]
+    extension = "html"
+    termstart = codestart
+    termend = codeend
+    out_width = nothing
+    out_height = nothing
+    fig_pos = nothing
+    fig_env = nothing
+    template = nothing
+    stylesheet = nothing
+    highlight_theme = nothing
+end
```

Comment:
Maybe we can do even more that this, like creating a macro that wraps `Base.@kwdef` and automatically defines some fields like `mimetypes` with default value if not specified, and do some assertion, etc, but we can do that later.

Reply:
That's actually not too bad an idea! I like it